File: src/main/scala/org/xbery/artbeams/products/admin/ProductForm.scala
package org.xbery.artbeams.products.admin

import net.formio.{Field, FormMapping, Forms}
import org.xbery.artbeams.common.form.ScalaForms
import org.xbery.artbeams.products.domain.EditedProduct

/**
  * Product edit form.
  * @author Radek Beran
  */
class ProductForm {
  lazy val definition: FormMapping[EditedProduct] = {
    Forms.basic(classOf[EditedProduct], "product")
      .field("id", Field.HIDDEN)
      .field("slug", Field.TEXT)
      .field("title", Field.TEXT)
      .field("fileName", Field.TEXT)
      .build(ScalaForms.CzConfig)
  }

}

File: src/main/kotlin/org/xbery/artbeams/products/admin/ProductForm.kt
package org.xbery.artbeams.products.admin

import net.formio.Field
import net.formio.FormMapping
import net.formio.Forms
import org.xbery.artbeams.common.form.FormUtils
import org.xbery.artbeams.products.domain.EditedProduct
import java.math.BigDecimal

/**
 * Product edit form.
 * @author Radek Beran
 */
open class ProductForm {
    companion object {
        val definition: FormMapping<EditedProduct> =
            Forms.basic(EditedProduct::class.java, "product")
                .field<String>("id", Field.HIDDEN)
                .field<String>("slug", Field.TEXT)
                .field<String>("title", Field.TEXT)
                .field<String?>("subtitle", Field.TEXT)
                .field<String>("fileName", Field.TEXT)
                .field<String?>("listingImage", Field.TEXT)
                .field<String?>("image", Field.TEXT)
                .field<String?>("confirmationMailingGroupId", Field.TEXT)
                .field<String?>("mailingGroupId", Field.TEXT)
                .field<BigDecimal?>("priceRegularAmount", Field.TEXT)
                .field<BigDecimal?>("priceDiscountedAmount", Field.TEXT)
                .build(FormUtils.CZ_CONFIG)
    }
}
